core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -150,19 +150,22 @@ class TileMap {
     }
 
 
-    fun getViewableTiles(position: Vector2, sightDistance: Int, ignoreCurrentTileHeight:Boolean=false): MutableList<TileInfo> {
-        val viewableTiles = getTilesInDistance(position, 1).toMutableList()
-        val currentTileHeight = if(ignoreCurrentTileHeight) 0 else get(position).getHeight()
-
-        for (i in 1..sightDistance) { // in each layer,
-            // This is so we don't use tiles in the same distance to "see over",
-            // that is to say, the "viewableTiles.contains(it) check will return false for neighbors from the same distance
-            val tilesToAddInDistanceI = ArrayList<TileInfo>()
-
-            for (tile in getTilesAtDistance(position, i)) { // for each tile in that layer,
-                val targetTileHeight = tile.getHeight()
-
-                /*
+    fun getViewableTiles(position: Vector2, sightDistance: Int, ignoreCurrentTileHeight: Boolean = false): MutableList<TileInfo> {
+        if (ignoreCurrentTileHeight) {
+            return getTilesInDistance(position, sightDistance).toMutableList()
+        } else {
```

Comment:
Please take this out of the 'else', since the previous 'if' ends in a 'return'
In other words, let's have a Straight Left Critical path

Reply:
You know what, I have no further comments, and this is very minor, so I'll deal with that